viewing_party/party.py

```diff
@@ -1,23 +1,153 @@
 # ------------- WAVE 1 --------------------
 
+# from imp import new_module
+# from turtle import title
+# from os import remove
+
+from codecs import unicode_escape_decode
+
+
 def create_movie(title, genre, rating):
-    pass
+    new_movie = {"title": title,
+            "genre": genre,
+            "rating": rating}
+    
+    if new_movie["genre"] == None:
+        return None
+    if new_movie["rating"] == None:
+        return None
+    if new_movie["title"] == None:
+        return None
+    return new_movie
+
+def add_to_watched(user_data, movie):
+    user_data = {}
+    user_data["watched"] = [movie]
+    return user_data
+
+def add_to_watchlist(user_data, movie):
+    user_data = {}
+    user_data["watchlist"] = [movie]
+    return user_data
+
+def watch_movie(user_data, title):
+    for movie in user_data["watchlist"]:
+        if movie["title"] == title:
+            user_data["watched"].append(movie)
+            user_data["watchlist"].remove(movie)
+            return user_data            
+    return user_data
+
 
 # -----------------------------------------
 # ------------- WAVE 2 --------------------
 # -----------------------------------------
+def get_watched_avg_rating(user_data):
+    rating = []
+    sum = 0.0
+
+    if not user_data["watched"]:
+        return sum
+    for i in user_data["watched"]: #4.0, 4.8
+        if i["rating"] != None: 
+            sum += i["rating"] #4.0 + 4.8 = 8.8 
+            rating.append(i["rating"]) #[4.0, 4.8]          
+    return sum/len(rating)
+
+
+def get_most_watched_genre(user_data): 
+    fav_genre = []
+    counter = 0
+    top_genre = fav_genre
+
+    for i in user_data["watched"]:
+        fav_genre.append(i["genre"])
+    
+    for genre in fav_genre:
+        curr_freq = fav_genre.count(genre)
+        if(curr_freq > counter):
+            counter = curr_freq
+            top_genre = genre
+            return top_genre
+        elif fav_genre == []:
+            return None
 
 
 # -----------------------------------------
 # ------------- WAVE 3 --------------------
 # -----------------------------------------
+def get_unique_watched(user_data):
+    list_of_movies = []
+    all_friend_movies = []
+
+    for friend in user_data["friends"]:
+        for movie in friend["watched"]:
+            if movie not in all_friend_movies:
+                all_friend_movies.append(movie)
+
+    for movie in user_data["watched"]:
+        if movie not in all_friend_movies:
+            list_of_movies.append(movie)
+    return list_of_movies    
+    
+def get_friends_unique_watched(user_data):
+    user_movies = []
+    unique_friend_movies = []
+
+    for movie in user_data["watched"]:
+        if movie not in user_movies:
+            user_movies.append(movie)

```

Comment:
Minor thing: Not sure I see the benefit of copying `user_data["watched"]` here to another list, when it is not being edited and thus can work just fine on line 103.